src/main/java/net/revelc/code/formatter/ValidateMojo.java

```diff
@@ -56,11 +77,24 @@ protected void doFormatFile(final File file, final ResultCollector rc, final Pro
         super.doFormatFile(file, rc, hashCache, basedirPath, true);
 
         if (rc.successCount != 0) {
-            throw new MojoFailureException("File '" + file
-                    + "' has not been previously formatted.  Please format file and commit before running validation!");
+            String errorMessage = String.format(
+                    "File '%s' has not been previously formatted. Please format file (for example by invoking `%s`) and commit before running validation!",
+                    file, formatCommand());
+            throw new MojoFailureException(errorMessage);
         }
         if (rc.failCount != 0) {
-            throw new MojoExecutionException("Error formating '" + file + "' ");
+            throw new MojoExecutionException("Error formatting '" + file + "' ");
+        }
+    }
+
+    private String formatCommand() {
+        String validateInvocation = String.format("mvn %s:%s:%s:%s@%s", mojoGroupId, mojoArtifactId, mojoVersion,
+                FORMAT_MOJO_NAME, mojoExecution.getExecutionId());
+        boolean isMultiModule = mavenSession.getAllProjects().size() > 1;
+        if (isMultiModule) {
+            return String.format("%s -f %s", validateInvocation, mavenProject.getBasedir());
+        } else {
+            return validateInvocation;
         }
```

Comment:
This would place the format goal before the `-f basedir/pom.xml`. Although this may work, `mvn --help` says, options, like `-f basedir/pom.xml` should be placed before the goals/phases. It'd be good to follow that consistently.

```suggestion
        return String.format("mvn%s %s", isMultiModule ? " -f " + mavenProject.getBasedir() : "", validateInvocation);
```

(This suggestion assumes 'mvn ' is removed from the front of `validateInvocation`.)

Reply:
Applied